app/src/main/java/lang_import/org/app/ArticleActivity.kt

```diff
@@ -36,18 +38,18 @@ class ArticleActivity : AppCompatActivity() {
         //TODO add title for ArticleActivity
         setTitle(intent.extras.getString("title"))
         status = "loading..."
-        importLang(clearText(intent.extras.getString("discript"))).whenComplete { readedTxt, ex ->
-            status = "preparing..."
-            if (ex != null) {
-                Log.e("import", "translate", ex)
-                return@whenComplete
-            }
-            val content = "<html><body>${readedTxt}<br/><br/><br/><br/></body></html>"
-            runOnUiThread {
-                webView.settings.javaScriptEnabled = false
-                webView.loadDataWithBaseURL(null, content, "text/html", "UTF-8", null)
+        launch {
+            val res = async { importLang(clearText(intent.extras.getString("discript"))) }.await()
+            fun preparing(res: String) {
```

Comment:
Why here a function that just will be invoked immediately after declaration? Can it be inlined?